src/Tempest/Database/src/Builder/ModelQueryBuilder.php

```diff
@@ -11,8 +11,9 @@
 
 /**
  * @template TModelClass of DatabaseModel
+ * @phpstan-ignore-next-line because this model query builder is extended
  */
-final class ModelQueryBuilder
+class ModelQueryBuilder
```

Comment:
We'll need to extract an interface + trait combo, we usually don't rely on inheritance :)